dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py

```diff
@@ -0,0 +1,753 @@
+        .rename(columns={"index": "job_id"})
+    )
+
+    return skill_measures_df, occs_measures_df, inds_measures_df, soc_name_dict
+
+
+def merge_green_measures(
+    skill_measures_df,
+    occs_measures_df,
+    inds_measures_df,
+    soc_name_dict,
+    job_id_col="job_id",
+):
+    soc_2020_6_dict = soc_name_dict["soc_2020_6"]
+    soc_2020_4_dict = soc_name_dict["soc_2020_4"]
+
+    all_green_measures_df = pd.merge(
+        skill_measures_df, occs_measures_df, how="outer", on=job_id_col
+    )
+    all_green_measures_df = pd.merge(
+        all_green_measures_df, inds_measures_df, how="outer", on=job_id_col
+    )
+    # replace float with 0
+    all_green_measures_df = all_green_measures_df.fillna("")
+    all_green_measures_df["NUM_GREEN_ENTS"] = all_green_measures_df["GREEN_ENTS"].apply(
+        len
+    )
+    # Separate out the SOC columns
+    for soc_columns in ["SOC_2020_EXT", "SOC_2020", "SOC_2010", "name"]:
+        all_green_measures_df[soc_columns] = all_green_measures_df["SOC"].apply(
+            lambda x: x[soc_columns] if x else None
+        )
+    all_green_measures_df.drop(columns=["SOC"], inplace=True)
+
+    all_green_measures_df.rename(
+        columns={"name": "SOC_names", job_id_col: "job_id"}, inplace=True
+    )
+    all_green_measures_df["SOC_2020_name"] = all_green_measures_df["SOC_2020"].map(
+        soc_2020_4_dict
+    )
+    all_green_measures_df["SOC_2020_EXT_name"] = all_green_measures_df[
+        "SOC_2020_EXT"
+    ].map(soc_2020_6_dict)
+
+    all_green_measures_df.rename(columns={})
+    all_green_measures_df.replace("", np.nan, inplace=True)
+
+    # weird thing in industry measures. 3 times
+    all_green_measures_df = all_green_measures_df[
+        all_green_measures_df["INDUSTRY GHG PER UNIT EMISSIONS"] != ":"
+    ]
+
+    # For multiskills the format is different in ENTS - separate them out
+    # ENTS: [[['research'], 'SKILL'], [['This is a cut up', 'a cut up sentence'], 'MULTISKILL']]
+    # to [[['research'], 'SKILL'], [['This is a cut up'], 'SKILL'], [['a cut up sentence'], 'SKILL']]
+
+    def separate_multiskill_ents(entlist):
+        separate_ents = []
+        for ent in entlist:
+            if ent[1] == "MULTISKILL":
+                for sep_ent in ent[0]:
+                    separate_ents.append(
+                        [[sep_ent], "MULTISKILL"]
+                    )  # The format it is expecting
+            else:
+                separate_ents.append(ent)
+        return separate_ents
+
+    all_green_measures_df["ENTS"] = all_green_measures_df["ENTS"].apply(
+        lambda x: separate_multiskill_ents(x) if isinstance(x, list) else x
+    )
+
+    print(f"There are {len(all_green_measures_df)} rows in the merged data")
+    print(f"There are {all_green_measures_df['job_id'].nunique()} unique job ids")
+
+    return all_green_measures_df
+
+
+def add_salaries(all_green_measures_df, job_id_col="job_id"):
+    salary_information = get_mixed_ojo_salaries_sample()
```

Comment:
we're going to need to remember to update these getters when we use the large sample

Reply:
omg very good shout! I'm going to add an argument from a config to help with this